components/process/provider.go

```diff
@@ -6,14 +6,16 @@ type Provider struct {
 }
 
 type Spec struct {
-	Directory   string            `json:"directory"`
-	Program     string            `json:"program"`
-	Arguments   []string          `json:"arguments"`
-	Environment map[string]string `json:"environment"`
+	Directory                  string            `json:"directory"`
+	Program                    string            `json:"program"`
+	Arguments                  []string          `json:"arguments"`
+	Environment                map[string]string `json:"environment"`
+	ShutdownGracePeriodSeconds int               `json:"shutdownGracePeriodSeconds"`
```

Comment:
This is a mouthful. Why not call it `shutdownTimeout`?

Also: Why seconds and not milliseconds? Seems like 200 ms would be more than enough grace period for many use cases.

Reply:
I basically took the name from Kubernetes which calls it the same thing [0]. `shutdownTimeout` sounds a bit like the maximum amount of time that the process should be allowed to run for.

There aren't any use cases which I can think of where 200ms would be better than 0 or 1 second. If a process intercepts the signal it is presumably going to do some I/O which I would always have more than a 1 second timeout for. For context, the default Kubernetes grace period is 30 seconds.

[0] https://kubernetes.io/docs/concepts/containers/container-lifecycle-hooks/#hook-handler-execution